### PR TITLE
[Artwork] Implement share modal functionality

### PR DESCRIPTION
**Pull request by damassi**
Addresses: https://artsyproduct.atlassian.net/secure/RapidBoard.jspa?rapidView=57&projectKey=DISCO&modal=detail&selectedIssue=DISCO-540&quickFilter=110

Wires up proper share functionality as well as refactors the structure a bit, also some light visual QA and swaps in new icons. 

save

**TODO:**
- Will follow up with second PR covering tests. 
- Need to wire up "Bell" icon for auction artworks

### CODE REVIEW

**Comment by artsyit** [comment]
Deploy preview for *artsy-reaction* ready!

Built with commit 2bd0e109c7b51361f1de3edd40e568d4272f17ba

https://deploy-preview-1718--artsy-reaction.netlify.com

File: src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkSharePanel.tsx [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+  graphql`
+    fragment ArtworkSharePanel_artwork on Artwork {
+      href
+      images {
+        url
+      }
+      imageDescription: description
+      title
+      date
+      artists {
```
Comment (mzikherman) [inline src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkSharePanel.tsx]: I didn't notice these fields being used in this component anywhere.

Comment (damassi) [reply src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkSharePanel.tsx]: Ahh good eye -- i just realized i need to wire up the proper description using the helper laid out in force

Comment (damassi) [reply src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkSharePanel.tsx]: Actually, can remove these and go with `meta { description }` instead

Comment (mzikherman) [reply src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkSharePanel.tsx]: The meta tag one in MP is a tad more verbose than the specific helper in Force for the shareable component.

I think it's minor, but if the length is an issue for something shareable specifically, can always implement that in MP (such as `meta { description(format: "short") }` kinda thing, where the 'short' version doesn't include medium and dimensions, kinda thing.

Comment (damassi) [reply src/Apps/Artwork/Components/ArtworkImageBrowser/ArtworkSharePanel.tsx]: @mzikherman There’s actually a character cap (no `format`):
```
type ArtworkMeta {
  description(limit: Int = 155): String
  image: String
  title: String
}
```

155 characters seems ok, yah?

**Pull request merged**